flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1485,6 +1485,8 @@ public void deliverOperatorEvent(OperatorID operator, SerializedValue<OperatorEv
         if (invokable instanceof CoordinatedTask) {
             try {
                 ((CoordinatedTask) invokable).dispatchOperatorEvent(operator, evt);
+            } catch (TaskNotRunningException e) {
+                throw e;
```

Comment:
Can we verify `currentState != ExecutionState.RUNNING && currentState != ExecutionState.INITIALIZING` here?

Reply:
It is possible for the `dispatchOperatorEvent` method to throw `TaskNotRunningException` when the task's mailbox has been closed while the execution state has not been set to values other than `RUNNING` yet. Thus we cannot add the verification mentioned above.